### PR TITLE
Enable l1-output-number conformance test for Go

### PR DESCRIPTION
**Pull request by Frassle**
This required fixing `__convert` in Go programgen to handle expressions that needed to result in input-y values. This also fixed the "single-or-none" codegen test.

### CODE REVIEW

File: pkg/codegen/go/gen_program_expressions.go [hunk 1]
```diff
@@ -235,7 +238,34 @@ func (g *generator) GenFunctionCallExpression(w io.Writer, expr *model.FunctionC
...
+				typeName := g.argumentTypeName(from, to, isOutput)
+				// IDOutput has a special case where it can be converted to a string
+				var isID bool
+				if expr, ok := from.(*model.ScopeTraversalExpression); ok {
+					last := expr.Traversal[len(expr.Traversal)-1]
+					if attr, ok := last.(hcl.TraverseAttr); ok && attr.Name == "id" {
+						isID = true
+					}
+				}
+				if expr, ok := from.(*model.RelativeTraversalExpression); ok {
```
Comment (lunaris) [inline pkg/codegen/go/gen_program_expressions.go:254]: Assuming these clauses are mutually exclusive, is this better as an `else if` or a type `switch`?

File: pkg/codegen/go/gen_program_test/batch1/gen_program_test.go [hunk 2]
```diff
@@ -26,5 +26,5 @@ import (
 
 func TestGenerateProgram(t *testing.T) {
 	os.Chdir("../../../go") // chdir into codegen/go
-	codegen.GenerateProgramBatchTest(t, test.ProgramTestBatch(1, 6))
+	codegen.GenerateProgramBatchTest(t, test.PulumiPulumiProgramTests) //test.ProgramTestBatch(1, 6))
```
Comment (lunaris) [inline pkg/codegen/go/gen_program_test/batch1/gen_program_test.go:29]: Accidental?

**Review by lunaris — APPROVED** [review]

File: pkg/codegen/go/gen_program_test/batch1/gen_program_test.go (hunk 2 above)
Comment (Frassle) [reply pkg/codegen/go/gen_program_test/batch1/gen_program_test.go:29]: Yup will revert

File: pkg/codegen/go/gen_program_expressions.go (hunk 1 above)
Comment (Frassle) [reply pkg/codegen/go/gen_program_expressions.go:254]: Yeh can switch it

**Pull request merged**